WordPress/Classes/ViewRelated/Post/PostEditorState.swift

```diff
@@ -240,21 +240,69 @@ public class PostEditorStateContext {
         self.userCanPublish = userCanPublish
         self.currentPublishDate = publishDate
         self.delegate = delegate
-        self.action = PostEditorStateContext.initialAction(for: originalPostStatus, userCanPublish: userCanPublish)
+        self.action = PostEditorStateContext.initialAction(for: originalPostStatus, publishDate: publishDate, userCanPublish: userCanPublish)
     }
 
-    private static func initialAction(for originalPostStatus: BasePost.Status?, userCanPublish: Bool) -> PostEditorAction {
+    private static func initialAction(for originalPostStatus: BasePost.Status?, publishDate: Date?, userCanPublish: Bool) -> PostEditorAction {
         // We assume an initial status of draft if none is set
         let newPostStatus = originalPostStatus ?? .draft
 
-        return action(for: originalPostStatus, newPostStatus: newPostStatus, userCanPublish: userCanPublish)
+        return action(for: originalPostStatus, newPostStatus: newPostStatus, publishDate: publishDate, userCanPublish: userCanPublish)
     }
 
     private static func action(
         for originalPostStatus: BasePost.Status?,
         newPostStatus: BasePost.Status,
-        userCanPublish: Bool) -> PostEditorAction {
+        publishDate: Date?,
+        userCanPublish: Bool
+    ) -> PostEditorAction {
+        if RemoteFeatureFlag.syncPublishing.enabled() {
+            return action(postStatus: originalPostStatus ?? .draft, publishDate: publishDate, userCanPublish: userCanPublish)
+        } else {
+            return _action(for: originalPostStatus, newPostStatus: newPostStatus, userCanPublish: userCanPublish)
+        }
+    }
+
+    static func action(for revision: AbstractPost) -> PostEditorAction {
+        action(
+            postStatus: revision.original().status ?? .draft,
+            publishDate: revision.dateCreated,
+            userCanPublish: revision.blog.capabilities != nil ? revision.blog.isPublishingPostsAllowed() : true
+        )
+    }
+
+    private static func action(
+        postStatus: BasePost.Status,
+        publishDate: Date?,
+        userCanPublish: Bool
+    ) -> PostEditorAction {
+        func makePublishAction() -> PostEditorAction {
+            guard userCanPublish else {
+                return .submitForReview
+            }
+            guard let publishDate else {
+                return .publish
+            }
+            return publishDate > .now ? .schedule : .publish
+        }
+        switch postStatus {
+        case .draft:
+            return makePublishAction()
+        case .pending:
+            return .update
```

Comment:
This will be a good place to allow admins to publish pending posts (according to spec-a1), but not in the scope of this PR.